src/library_fs.js

```diff
@@ -838,7 +838,7 @@ mergeInto(LibraryManager.library, {
       if (!link.node_ops.readlink) {
         throw new FS.ErrnoError(ERRNO_CODES.EINVAL);
       }
-      return link.node_ops.readlink(link);
+      return PATH.resolve('.', link.node_ops.readlink(link));
```

Comment:
@kripken This is just for the tests to give consistent result. F.e. a readlink could give back ../test/../there! (see  tests/unistd/links.out ), but in node we now call PATH.join, which internally normalizes it. To be consistent we need to normalize it here as well, otherwise we get inconsistent results between running against memfs & nodefs.